### PR TITLE
feat: add title to strategy

### PR DESCRIPTION
**Pull request by andreas-unleash**
Adds title column to strategies, feature_strategies and features_view in the db
Updates model/schemas
## About the changes

Closes # [1-855](https://linear.app/unleash/issue/1-855/allow-for-title-on-strategy-backend)

### CODE REVIEW

File: src/lib/openapi/spec/strategy-schema.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ export const strategySchema = {
         'parameters',
     ],
     properties: {
+        title: {
```
Comment (kwasniew) [inline src/lib/openapi/spec/strategy-schema.ts:16]: I'd add descriptions and examples in this or next PR

File: src/lib/services/strategy-schema.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ const strategySchema = joi
     .object()
     .keys({
         name: nameType,
+        title: joi.string().default('').optional(),
```
Comment (kwasniew) [inline src/lib/services/strategy-schema.ts:8]: so do we want to persist empty string for new strategies? 

Also is we have optional and default isn't default enough?

**Review by kwasniew — APPROVED** [review]
A few minor comments inside

File: src/lib/services/strategy-schema.ts (hunk 2 above)
Comment (andreas-unleash) [reply src/lib/services/strategy-schema.ts:8]: Actually, I will make it the same as description here, thanks for catching

File: src/lib/openapi/spec/strategy-schema.ts (hunk 1 above)
Comment (andreas-unleash) [reply src/lib/openapi/spec/strategy-schema.ts:16]: Good call, I will fix the schema since I am here

Comment (andreas-unleash) [reply src/lib/openapi/spec/strategy-schema.ts:16]: Good call.

**Comment by sjaanus** [comment]
Can we update e2e tests also to check that title gets saved on creation and on update?

**Pull request merged**